crates/i2c/src/main.rs

```diff
@@ -168,7 +168,7 @@ impl TryFrom<I2cArgs> for I2cConfiguration {
 
         let devices = AdapterConfig::try_from(args.device_list.trim())?;
         Ok(I2cConfiguration {
-            socket_path: args.socket_path.trim().to_string(),
```

Comment:
This is something I don't know if it still makes sense or not. It all depends on how clap would parse a PathBuf argument and whether it trims the unnecessary spaces. Anyone familiar with clap that knows the answer?

Reply:
I think it is even wrong to trim... Leading and trailing spaces are perfectly fine - though odd - filenames, right? But maybe remove it in a standalone commit.

Reply:
> I think it is even wrong to trim... Leading and trailing spaces are perfectly fine - though odd - filenames, right? But maybe remove it in a standalone commit.

The problem is there are tests for at least rng and i2c that checks that socket path argument has it trailing spaces trimmed which is ... weird?

Reply:
Yeah. Saw that. I would drop them. Maybe we can double check with @vireshk.

Reply:
Just my 2c, I changed to `PathBuf` also for the vhost-video PR, and ended up removing the `trim()` part and dropping the check in the tests I had. It works perfectly fine without triming, and checking if the filename is parsed correctly should be enough.

Reply:
I think I got into some issues earlier with the extra spaces. Look at https://github.com/rust-vmm/vhost-device/commit/b29a5e228583735a9cfd68e86559830fb4054c59.

Reply:
The spaces depend on how the argv is populated, if the shell gets `cargo run -- [..] --socket-path "    /path"` from a user then yes there will be spaces in the value, and it should result in an invalid path error. Look at how core utils handle this case:

```shell
$ ls "   /tmp"
ls: cannot access '   /tmp': No such file or directory
```

Indeed white space characters are valid for paths and it doesn't always hold that `path_string.trim() == path_string.as_str()`:

> 
>  3.266 Pathname
> 
> A character string that is used to identify a file. In the context of IEEE Std 1003.1-2001, a pathname consists of, at most, {PATH_MAX} bytes, including the terminating null byte. It has an optional beginning slash, followed by zero or more filenames separated by slashes. A pathname may optionally contain one or more trailing slashes. Multiple successive slashes are considered to be the same as one slash.  [^posix1]


[^posix1]: https://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap03.html#tag_03_266